components/map.jsx

```diff
@@ -1,68 +1,75 @@
 var React = require('react');
+var mapboxId = process.env.MAPBOX_MAP_ID || 'alicoding.ldmhe4f3';
+var accessToken = process.env.MAPBOX_ACCESS_TOKEN || 'pk.eyJ1IjoiYWxpY29kaW5nIiwiYSI6Il90WlNFdE0ifQ.QGGdXGA_2QH-6ujyZE2oSg';
+var teachAPI = require('../lib/teach-api');
 
-var Map = React.createClass({
+function geoJSONit(data) {
+  return data.map(function(i) {
+    return {
+      "geometry": {
+        "coordinates": [i.longitude*1, i.latitude*1],
+        "type": "Point"
+      },
+      "properties": {
+        "description": i.description,
+        "location": i.location,
+        "title": i.name
+      },
+      "type": "Feature"
+    }
+  });
+}
 
+var Map = React.createClass({
   propTypes: {
-    accessToken: React.PropTypes.string.isRequired,
-    mapId: React.PropTypes.string.isRequired,
     className: React.PropTypes.string.isRequired
   },
   componentDidMount: function() {
+    var that = this;
     require('mapbox.js'); // this will automatically attach to Window object.
-    this.map = L.mapbox.map(this.getDOMNode(), this.props.mapId, {
-      accessToken: this.props.accessToken
-    }).setView([43.597, -79.6139], 12);
-
-    var MapLayer = L.mapbox.featureLayer(this.mapId, {accessToken: this.props.accessToken}).addTo(this.map);
-    var geoJson = [{
-      // this feature is in the GeoJSON format: see geojson.org
-      // for the full specification
-      type: 'Feature',
-      geometry: {
-        type: 'Point',
-        // coordinates here are in longitude, latitude order because
-        // x, y is the standard for GeoJSON and many formats
-        coordinates: [-79.6139,
-          43.597
-        ]
-      },
-      properties: {
-        title: 'First Point',
-        description: 'Party description?',
-        "icon": {
-          "iconUrl": "/img/map-marker.svg",
-          "iconSize": [33, 33], // size of the icon
-          "iconAnchor": [15, 15] // point of the icon which will correspond to marker's location
-        }
+    require('leaflet.markercluster');
+    L.mapbox.accessToken = accessToken;
+    teachAPI.getAllData(function(err, data) {
+      if(err) {
+        console.log(err)
```

Comment:
Hmm we should probably figure out something more robust to do here, but can leave that for a later PR...